src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceFocusState?: boolean;
+
+  /**
+   * Force the button to be in the hover state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceHoverState?: boolean;
+
+  /**
+   * A hidden button will have no background
+   *
+   * You should configure the `hiddenActiveBackgroundColor` prop if you are
+   * using any custom colors.
+   */
+  hidden?: boolean;
+
+  /**
+   * Color to make the background for a hidden button that is active.
+   *
+   * This color should be darker than the `hoverBackgroundColor`.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenActiveBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Color to make the text for a focused hidden button.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenFocusedColor?: CSS.ColorProperty;
+
+  /**
+   * Either an icon to show to the left of the button text, or on it's own
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Size of the button
+   *
+   * Defaults to "default"
+   */
+  size?: "default" | "small" | "large";
+
+  /**
+   * CSS styles to apply
+   *
+   * @ignore
+   */
+  style?: React.CSSProperties;
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  backgroundColor = colors.silver.light,
+  children,
+  disabled = false,
+  disabledBackgroundColor = colors.silver.dark,
+  disabledTextColor = colors.grey.light,
+  fab = false,
+  forceActiveState = false,
+  forceFocusState = false,
+  forceHoverState = false,
+  hidden = false,
+  hiddenActiveBackgroundColor = colors.silver.dark,
+  hiddenFocusedColor = colors.blue.base,
+  hoverBackgroundColor = colors.silver.base,
+  icon,
+  size = "default",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  if (forceActiveState && forceHoverState) {
+    throw new TypeError(
+      "Do not force multiple properties at once, you will get indeterminiate behavior"
+    );
+  }
+
+  // We're storing this because it can be used for the `:active` style or when
+  // `forceActiveState` is used in Storybook
+  const activeProperties = {
+    ...(hidden
+      ? { backgroundColor: hiddenActiveBackgroundColor }
+      : {
+          // The `box-shadow` property is copied directly from Zeplin
+          boxShadow:
```

Comment:
In the docs, I made a function for getting box shadows since some of the button states have box shadows in common. Here's how it looks:

```js
import {colors} from '@apollo/space-kit';
import {transparentize} from 'polished';

const highlightColor = 
const shadowColor = colors.black.darker;
function getBoxShadow(state) {
  const isFocused = state === 'focused';
  return [
    state === 'active'
      ? `inset 0 2px 2px 0 ${transparentize(0.88, shadowColor)}`
      : `0 1px 4px 0 ${transparentize(0.92, shadowColor)}`,
    `inset 0 -1px 0 0 ${transparentize(0.95, shadowColor)}`,
    isFocused && `0 0 0 2px ${colors.blue.lightest}`,
    `inset 0 0 0 1px ${
      isFocused ? colors.blue.lighter : transparentize(0.8, shadowColor)
    }`
  ]
    .filter(Boolean)
    .toString();
}

<button
  css={{
    boxShadow: getBoxShadow(),
    ':hover': {
      boxShadow: getBoxShadow('hover')
    },
    ':active': {
      boxShadow: getBoxShadow('active')
    },
    ':focus': {
      boxShadow: getBoxShadow('focused')
    }
  }}
/>
```

Reply:
I would prefer to be able to copy-paste the values directly from the zeplin designs instead of doing this. Thanks for diving deep into this :)